### PR TITLE
feat: migrate /apps/routing-forms/forms to /routing

### LINKED ISSUE #17955: [CAL-4828] migrate /apps/routing-forms/forms to /routing
we want to move [https://app.cal.com/apps/routing-forms/forms](https://app.cal.com/apps/routing-forms/forms) → [https://app.cal.com/routing](https://app.cal.com/routing)

- [ ] move all files out of /apps/
- [ ] change URL path
- [ ] redirect old path to new path
- [ ] rename "Routing Forms" to "Routing"
- [ ] new icon: "split"

before:

CleanShot 2024-12-02 at 21 42 00@2x

after:

CleanShot 2024-12-02 at 21 44 32@2x

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4828](https://linear.app/calcom/issue/CAL-4828/migrate-appsrouting-formsforms-to-routing)</sub>

### PR DESCRIPTION
**Pull request by VK-RED**
## What does this PR do?

- Setup /routing page
- Added redirect from /apps/routing-forms/forms to /routing
- Renamed "Routing Forms" to "Routing" and changed the icon to split
- Fixes #17955
- Fixes CAL-4828

### Screencast

[Screencast from 2024-12-04 19-45-04.webm](https://github.com/user-attachments/assets/43aca372-fa69-4ebe-9dcb-015f4802afdd)

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: apps/web/middleware.ts [hunk 1]
```diff
@@ -41,6 +41,15 @@ const middleware = async (req: NextRequest): Promise<NextResponse<unknown>> => {
     }
   }
 
+  // redirect /apps/routing-forms/forms to /routing
```
Comment (Praashh) [inline apps/web/middleware.ts:44]:
```suggestion
```

File: apps/web/middleware.ts [hunk 2]
```diff
@@ -41,6 +41,15 @@ const middleware = async (req: NextRequest): Promise<NextResponse<unknown>> => {
...
+
+  if (url.pathname.startsWith("/apps/routing-forms/forms")) {
+    const redirectPathName = url.pathname.replace(/^\/apps\/routing-forms\/forms$/, "/routing");
+    const nextUrl = url.clone();
+    nextUrl.pathname = redirectPathName;
+    return NextResponse.redirect(nextUrl, { headers: requestHeaders });
+  }
+
```
Comment (Praashh) [inline apps/web/middleware.ts:52]: why are we creating a new URL and cloning it? why not just redirect it like 

```
if (url.pathname === "/apps/routing-forms/forms") {
  return NextResponse.redirect("/routing", { headers: requestHeaders });
 }
```

File: apps/web/middleware.ts (hunk 1 above)
Comment (VK-RED) [reply apps/web/middleware.ts]: removed the comment

**Review by Praashh — CHANGES_REQUESTED** [review]
e2e tests are failing, can you resolve this 🙏

**Comment by VK-RED** [comment]
> e2e tests are failing, can you resolve this 🙏

Yeah sure :+1:

**Review by Praashh — APPROVED** [review]
LGTM!!, @anikdhabal please give it a fina touch 🙏

**Comment by PeerRich** [comment]
are all URLs redirecting accordingly?

**Comment by PeerRich** [comment]
requested review by @hariombalhara

**Comment by emrysal** [comment]
@PeerRich Best to put in Changes Requested

Reviewed in @hariombalhara's stead, everything works as it is supposed to, however, although a bit vague from the reported feature request, I would assume this should happen also - not just the /forms page:

```
./apps/routing-forms/form-edit/[...id] -> ./routing/form-edit/[...id]
./apps/routing-forms/route-builder/[...id] -> ./routing/route-builder/[...id]
&
./apps/routing-forms/reporting/[...id] -> ./routing/reporting/[...id]
```

This does not happen yet, @PeerRich to confirm; if so: @VK-RED if you could also have a look at this in a follow-up?

**Comment by emrysal** [comment]
Hi @VK-RED I've decided to revert this change, the reason for this is that it's a (minor) additional overhead in middleware, is missing some key parts such as additional URLs & there should have been a code cleanup as part of this.

I'll talk with @hbjORbj to aid in this also as part of the appDir migration as it's an ideal migration usecase, will re-open the issue to continue collaborating.

**Comment by VK-RED** [comment]
@emrysal Sure will followup on this .. I thought we only needed to change /apps/routing -forms/routing to /routing . Will do this for all /apps/routing-forms routes too..

**Pull request merged**